idaes/apps/grid_integration/tracker.py

```diff
@@ -0,0 +1,261 @@
+import pandas as pd
+import pyomo.environ as pyo
+from collections import deque
+import os
+
+class Tracker:
+
+    def __init__(self, tracking_model_class, n_tracking_hour, solver, **kwarg):
+
+        '''
+        Initializes the tracker object.
+
+        Arguments:
+            tracking_model_class: the model object class for tracking
+            n_tracking_hour: number of implemented hours after each solve
+            solver: a Pyomo mathematical programming solver object
+            **kwarg: necessary arguments to initialize the selected model object
+
+        Returns:
+            None
+        '''
+
+        # create an instance
+        self.tracking_model_object = tracking_model_class(**kwarg)
+
+        # add flowsheet to model
+        self.model = pyo.ConcreteModel()
+        self.model.fs = pyo.Block()
+        b = self.tracking_model_object.create_model()
+        self.model.fs.transfer_attributes_from(b)
+
+        # get the power output
+        power_output_name = self.tracking_model_object.power_output
```

Comment:
This should throw a specific error if tracking_model_object does not have `power_output`.

Reply:
Or perhaps the model object should throw specific errors if `create_model`, etc. are not properly defined.